virtualizarr/readers/hdf/hdf.py

```diff
@@ -192,40 +205,19 @@ def open_virtual_dataset(
             filepath, fs_root=Path.cwd().as_uri()
         )
 
-        _drop_vars: list[Hashable] = (
+        _drop_vars: Iterable[str] = (
             [] if drop_variables is None else list(drop_variables)
         )
 
-        # TODO provide a way to drop a variable _before_ h5py attempts to inspect it?
-        virtual_vars = HDFVirtualBackend._virtual_vars_from_hdf(
-            path=filepath,
-            group=group,
-            reader_options=reader_options,
-        )
-
-        attrs = HDFVirtualBackend._get_group_attrs(
-            path=filepath, reader_options=reader_options, group=group
-        )
-        coordinates_attr = attrs.pop("coordinates", "")
-        coord_names = coordinates_attr.split()
-
-        fully_virtual_dataset = construct_fully_virtual_dataset(
-            virtual_vars=virtual_vars,
-            coord_names=coord_names,
-            attrs=attrs,
-        )
-
-        vds = construct_virtual_dataset(
-            fully_virtual_ds=fully_virtual_dataset,
+        manifest_store = HDFVirtualBackend._create_manifest_store(
             filepath=filepath,
+            drop_variables=_drop_vars,
             group=group,
-            loadable_variables=loadable_variables,
-            reader_options=reader_options,
-            indexes=indexes,
-            decode_times=decode_times,
         )
-
-        return vds.drop_vars(_drop_vars)
+        ds = manifest_store.to_virtual_dataset(
+            loadable_variables=loadable_variables, decode_times=decode_times
+        )
```

Comment:
do you need to pass indexes in here?

Reply:
IIUC we don't have support for processing indexes yet https://github.com/zarr-developers/VirtualiZarr/blob/develop/virtualizarr/xarray.py#L56-L57

Reply:
IMO it'd be better to raise the not supported error than the silently not do anything if a user passes in an argument to indexes

Reply:
Agreed

Reply:
@maxrjones Addressed by https://github.com/zarr-developers/VirtualiZarr/pull/542/commits/a3bfca98faf405acdc5982517fa172c232d85d03